examples/bri-3/src/bri/transactions/agents/mockTransactionStorage.agent.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+import { Transaction } from '../models/transaction';
+
+@Injectable()
+export class MockTransactionStorageAgent {
+  private transactionsStore: Transaction[] = [];
+
+  async getTransactionById(id: string): Promise<Transaction> {
+    const transaction = this.transactionsStore.find((bp) => bp.id === id);
+    if (!transaction) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+    return transaction;
+  }
+
+  async createNewTransaction(transaction: Transaction): Promise<Transaction> {
+    const createdBp = new Transaction(
+      transaction.id,
+      transaction.nonce,
+      transaction.workflowInstanceId,
+      transaction.workstepInstanceId,
+      null, // TODO: transactionModel.from once BpiAccount in the prisma schema,
+      null, // TODO: transactionModel.to once BpiAccount in the prisma schema,
+      transaction.payload,
+      transaction.signature,
+      transaction.status,
+    );
+
+    this.transactionsStore.push(createdBp);
+
+    return Promise.resolve(createdBp);
+  }
+
+  async updateTransaction(transaction: Transaction): Promise<Transaction> {
+    const txToUpdate = this.transactionsStore.find(
+      (bp) => bp.id === transaction.id,
+    );
+    Object.assign(txToUpdate, transaction);
```

Comment:
since the array is never updated, the getter calls from array will still return the previous value and not the updated one. Please update in the array also

Reply:
Then the update test would fail, wouldn't it? txToUpdate is storing a reference to the actual object, so updating it actually updates the underlying array.

Reply:
yeah, that's why it has been failing for me

Reply:
So this is an issue after the introduction of generic mapper and is solved on your branch, if i understand correctly